framework/include/parser/CommandLine.h

```diff
@@ -116,6 +116,17 @@ class CommandLine
     return unused;
   }
 
+  enum class HitCmpResult
+  {
+    GLOBAL,
+    MAIN,
+    MATCH,
+    WRONG
+  };
+
+  static HitCmpResult
```

Comment:
This function definitely would benefic from a good doc comment explaining what it does and how to use it.  Something like:

`Determines whether the given arg string follows the patterns for a HIT parameter that is applicable: globally to all master and multiapp subapps, only to the main/master app, only to the particular specified multiapp/subapp (via appname and appnum), or to another different multiapp/subapp.`